### ISSUE TITLE
Prepare a version with semantic event stream

### ISSUE THREAD

**Issue by samreid** [phetsims/capacitor-lab-basics#218]
At today's status meeting, @kathy-phet said a study is coming up that will require Capacitor Lab: Basics and the semantic event stream.  It is unclear whether we should use the previous version (used in another study) or publish a new version.

I told @kathy-phet that I'm not too familiar with the status of Capacitor Lab Basics, and that it would be helpful to have @jonathanolson's help in development.

@kathy-phet said we will soon need to dedicate work to getting this ready.   Cross assigning to @samreid and @jonathanolson to check status and assigning to @kathy-phet to get more information about timeline, proposed usage, etc.

**Comment by jonathanolson** [comment]
I'll be available to assist (and even if not necessary, it may be helpful for me to better understand the current approach to instrumentation).

**Comment by samreid** [comment]
@kathy-phet Are Ido and his team aware that the data stream from the current version will not be compatible with the data stream from the previous version? (They would likely need to update/rewrite their parser).

@jonathanolson and I are facing this question as we review the sim instrumentation because we are finding places where we recommend to update a tandem name in the code but are hesitant to do so if it will create more work for Ido and his team.  Also, it may be suboptimal if Ido has to deal with a 3rd schema in the future when we have the Stable PhET-iO design meeting for this simulation and adjust the names yet again.

**Comment by samreid** [comment]
I build a version and checked each wrapper briefly.  Everything seems to be in reasonable shape aside from https://github.com/phetsims/capacitor-lab-basics/issues/221 which doesn't seem to be a blocking issue.

**Comment by samreid** [comment]
I'm halfway through skimming the tandem and var names.

**Comment by samreid** [comment]
I passed through all of the cases, updated several and added several TODOs.

**Comment by samreid** [comment]
One commit is here: https://github.com/phetsims/capacitor-lab-basics/commit/fc356c7a2705cb0f777c22c4f7bea9f791990599

**Comment by samreid** [comment]
@kathy-phet said it would be good to give an initial version to collaborators by the end of next week.  I mentioned that I'd still like to look through each file, @kathy-phet agreed that is reasonable as long as we make sure it doesn't take too long.

**Comment by samreid** [comment]
The state wrapper, data stream and instance proxies seem functional.  Instance proxies seems like far too many things are instrumented, weighing in at 242 instances.  This will hopefully be pruned at the stable PhET-iO design meeting.

I'd like to take another look at the state and console phetioIDs, then this simulation should be ready for @phet-steele.  It's unclear how much testing this sim should receive.  Do we want to fully test and publish a new production version?  Tagging @ariel-phet so he's aware of this on the testing horizon for next week.  Tagging @kathy-phet to discuss the level of testing requested.

**Comment by kathy-phet** [comment]
Yes, let’s do good testing on this and publish  as available. Although not necessarily final instrumentation.

First round of testing should be sanity check. Then more obscure usage.

When you have a good link, can you send it with info on instance proxies to tanja at stanford? CC’d

Thx, and excuse brevity,
K





Sent from my iPhone

On Jan 12, 2018, at 2:18 PM, Sam Reid <notifications@github.com<mailto:notifications@github.com>> wrote:


The state wrapper, data stream and instance proxies seem functional. Instance proxies seems like far too many things are instrumented, weighing in at 242 instances. This will hopefully be pruned at the stable PhET-iO design meeting.

I'd like to take another look at the state and console phetioIDs, then this simulation should be ready for @phet-steele<https://github.com/phet-steele>. It's unclear how much testing this sim should receive. Do we want to fully test and publish a new production version? Tagging @ariel-phet<https://github.com/ariel-phet> so he's aware of this on the testing horizon for next week. Tagging @kathy-phet<https://github.com/kathy-phet> to discuss the level of testing requested.

**Comment by samreid** [comment]
Ranges for many number properties are incorrect.

**Comment by samreid** [comment]
I published a dev version here:

https://www.colorado.edu/physics/phet/dev/html/capacitor-lab-basics/1.6.0-dev.3/phet-io/wrappers/index/

I'll create a QA issue for testing.

**Comment by samreid** [comment]
This is out for testing in the aforementioned issue.  Unassigning for now.

**Comment by samreid** [comment]
Out for testing in https://github.com/phetsims/QA/issues/86
